mesonbuild/utils/universal.py

```diff
@@ -2411,22 +2411,22 @@ def is_base(self) -> bool:
         return self.type is OptionType.BASE
 
 
-def pickle_load(filename: str, object_name: str, object_type: T.Type[_PL]) -> _PL:
-    load_fail_msg = f'{object_name} file {filename!r} is corrupted. Try with a fresh build tree.'
+def pickle_load(filename: str, object_name: str, object_type: T.Type[_PL], suggest_reconfigure: bool = True) -> _PL:
+    load_fail_msg = f'{object_name} file {filename!r} is corrupted.'
```

Comment:
Hmm, it seems we also lost the unpickling error "Try with a fresh build tree" and that isn't replaced with extra_msg either.

Reply:
You'll get the more precise extra_msg instead. Unless you're already reconfiguring in which case there is no point in telling you to refresh because it's already what it's doing.

Reply:
Indeed, I misread that and you are indeed passing extra_msg in its place.